Add multi-pin and proration explainer (#54)
ccao-data/wiki: Residential/Multi-Card PINs/multi_card_explainer.md

> # Valuing Multi-PIN and Multi-Card Aggregations

Reviewer: **[Suggestion, blocking]** This is a good start! I think we could use even more detail, though. Some additional info I'd like to see in this doc:

1. Explanation and example for how 3+ card PINs get valued
2. Explanation and example for how we break ties in building square footage
3. Explanation why we exclude multicard sales from training

I'm also interested in the explanations being included in text, alongside the diagrammatic examples. The diagrams are helpful but for clarity and accessibility I think we should explain everything in text too.

This is a lot of new info, so let me know if you want to talk through a possible structure for a doc that includes all this info and we can outline it together.

Author: Took a shot at this. Let me know what you think

# Valuing Multi-Card PINs

## Introduction

Multi-card parcels are properties (PINs) containing more than one building, each represented individually
as a separate "card."

Our model is trained at the building level, using specific features, characteristics, and

> location data for each individual structure. This makes it difficult to accurately value parcels that contain multiple buildings, since the model is designed to assess one building at a time—and because multi-card PINs are excluded from the training data, the model has no exposure to these more complex cases

Reviewer: **[Nitpick, non-blocking]** Missing period:

```suggestion
location data for each individual structure. This makes it difficult to accurately value parcels that contain multiple buildings, since the model is designed to assess one building at a time—and because multi-card PINs are excluded from the training data, the model has no exposure to these more complex cases.
```

# Valuing Multi-Card PINs

## Introduction

Multi-card parcels are properties (PINs) containing more than one building, each represented individually
as a separate "card." We have two different methods for valuing these parcels, depending on the number of cards

> involved. For parcels with 2 or 3 cards, we use a new method that aggregates the total building square footage

Reviewer: **[Nitpick, non-blocking]** Broadly speaking, I don't think we need to refer to the "new" and "old" methods in this doc -- I think it's easier to understand if we just distinguish between the 2-3 card method and the 4+ card method.

```suggestion
involved. For parcels with 2 or 3 cards, we use a method that aggregates the total building square footage
```

> from all cards into a single "main" card and performs a single prediction. For parcels with 4 or more cards,

Reviewer: **[Nitpick, non-blocking]** Can we quickly add a note that we then divide this single prediction between the cards based on square footage?

> we continue to use the legacy method of summing individual card-level predictions. This document explains the

Reviewer: **[Nitpick, non-blocking]** Same here;

```suggestion
we sum individual card-level predictions to get the parcel-level prediction. This document explains the
```

framework behind these methods and the rationale for our approach.

Our model is trained at the building level, using specific features, characteristics, and
location data for each individual structure. This makes it difficult to accurately value
parcels that contain multiple buildings, since the model is designed to assess one building
at a time. Multi-card PINs are also excluded from the training data, so the model doesn't

> see examples of them during training.

Reviewer: **[Suggestion, non-blocking]** It feels to me like this paragraph is more general than the one that precedes it, so I wonder if we should swap them? Maybe keep `Multi-card parcels are properties (PINs) containing more than one building, each represented individually as a separate "card"` as the first sentence, but otherwise swap the two paragraphs?

![](model_single_card_assumption.PNG)

### Why We Exclude Multi-Card PINs from Training

Machine learning models work better when they are trained on data with
a consistent structure. Ensuring every single row in the training data is a single-card

> PIN helps the model learn the relationship between features and values more effectively.

Reviewer: **[Nitpick, non-blocking]** Slight clarification:

```suggestion
sale helps the model learn the relationship between features and values more effectively.
```

It doesn't have to account for variation between single-card and multi-card PINs. We've
confirmed this through testing, including multi-card PINs in the training data results
in a significant drop in performance for single-card PINs, which make up the overwhelming

> majority of residential PINs in cook county.

Author: It is at least verifiable in our [internal testing](https://github.com/ccao-data/model-res-avm/pull/328) looking at error metrics on an assessment set, which probably isn't polished enough to link out to:
| Metric | grouped single card preds | sf aggregation| % Drop |
|--------|---------|---------|--------|
| MdAPE  | 67.02   | 28.8    | 57%    |
| RMSE   | 879077  | 508793  | 42%    |

I am getting very nerd-sniped trying to answer precisely why I believe this is theoretically stronger. Maybe we can discuss out of band.

> majority of residential PINs in cook county.

Author: Made some edits here

> majority of residential PINs in cook county.

Reviewer: Thanks for the clarification, this all makes sense to me!

and improved performance for multi-card PINs.

![](model_multi_card_aggregation.PNG)

### Breaking Ties in Building Square Footage

When selecting which card should act as the "main" for aggregation, we apply the following logic:

1. Choose the card with the **largest building square footage**.
2. If there's a **tie**, we select the card with the **lowest card number** (e.g., Card 1 before Card 2).

**Example:**  
_If two buildings are both 1,200 sqft and one is Card 1 and the other is Card 2, we designate Card 1 as the main._

---

## Valuing 4+ Card PINs

For PINs with 4 or more cards, we continue using the previous method of summing individual card-level
predictions. These cases are rare and often unusually complex, sometimes resulting from administrative
anomalies in the underlying data. For example, a recorded sale might include several cards when, in
reality, one of those buildings has been split off into a new PIN that isn’t captured in our data.
Because of these inconsistencies—and the lack of a reliable theoretical foundation for aggregation
in such cases—these PINs are usually flagged for additional manual review. We don't yet have the

> confidence or clarity in the data structure to apply the same aggregation method used for 2–3 card PINs.

Author: Added
